File: problem-solving/baekjoon/2847_게임을_만든_동준이.py
# https://www.acmicpc.net/problem/2847
import sys

if __name__ == "__main__":
    n = int(sys.stdin.readline())
    scores = [int(sys.stdin.readline()) for _ in range(n)]
    result = 0 
    for i in range(n-2, -1, -1):
        if scores[i+1] <= scores[i]:
            diff = scores[i] - scores[i+1] + 1
            result += diff
            scores[i] -= diff
    print(result)

File: problem-solving/baekjoon/2847_게임을_만든_동준이.go
// https://www.acmicpc.net/problem/2847
package main

import (
	"bufio"
	"fmt"
	"os"
)

func main() {
	reader := bufio.NewReader(os.Stdin)
	writer := bufio.NewWriter(os.Stdout)
	defer writer.Flush()

	var n int
	fmt.Fscanln(reader, &n)
	var scores = make([]int, n)
	for i := 0; i < n; i++ {
		fmt.Fscanln(reader, &scores[i])
	}
	var result int
	for i := n - 2; i >= 0; i-- {
		if scores[i+1] <= scores[i] {
			diff := scores[i] - scores[i+1] + 1
			result += diff
			scores[i] -= diff
		}
	}
	fmt.Fprintln(writer, result)
}
